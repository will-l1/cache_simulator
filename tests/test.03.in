1024
65536
4
7 
519
15 
527
stats
