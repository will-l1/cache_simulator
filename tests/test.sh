#!/bin/sh

echo ======================================================
echo ====================== TEST $1 ========================
echo ======================================================
if timeout 10 ./$2/$3 < tests/test.$1.in > tests/test.$1.out; then 
  if diff -w tests/test.$1.out tests/test.$1.expected > /dev/null; then
    echo PASSED
  else
    echo FAILED
    echo ======
    echo ____Your_output____ __Expected_Output___
   #echo +++++++++++++++++++ ++++++++++++++++++++
    diff -y -W 80 tests/test.$1.out tests/test.$1.expected 
    echo =====================
    echo Your Output:
    echo ++++++++++++++++
    cat tests/test.$1.out
    echo =====================
    echo Expected Output:
    echo ++++++++++++++++
    cat tests/test.$1.expected
    exit 1
  fi
elif [ $? -eq 124 ]; then
  echo TIMEOUT
  exit 1
else 
  echo Abnormal program termination: the program crashed
  echo Exit code $?
  exit 1
fi
