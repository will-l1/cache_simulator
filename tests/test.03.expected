Loaded value [0x31bb7e7a43899248] @ address 0x00000007
Loaded value [0xf0b225047b14b95a] @ address 0x00000207
Loaded value [0xd2db0924ee69530a] @ address 0x0000000f
Loaded value [0x5f63842511bd4341] @ address 0x0000020f
Cache hits: 2, misses: 2 -- hit rate 50%
